### PR TITLE
Assignment 8 : Memory Management

### PR DESCRIPTION
**Pull request by shafiq005-solin**
# Description
This assignment is help us to learn about the memory organization and how it works. And what are the memories available and how it is organized and how to manage the memory with high efficiency. The concept of Garbage collection and Idisposable. 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or errors
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/assignment-8/IDisposableDemo/FileManager.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            catch (Exception error)
+            {
+                Console.WriteLine($"Some error occured, that is {error.Message}");
+            }
+
+            return "File not found";
+        }
+
+        /// <summary>
+        /// Colse all the open files
```
Comment (MithunSubramani) [inline src/assignment-8/IDisposableDemo/FileManager.cs:57]: check the typo.

File: src/assignment-8/IDisposableDemo/FileManager.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        /// <summary>
+        /// Read the content from the file
+        /// </summary>
+        /// <returns>Content from the file</returns>
+        public string FileRead()
+        {
+            this._fileWriter.Close();
+            try
+            {
+                StreamReader fileReader = new StreamReader(@"D:\CSharp-Assignments\src\assignment-8\IDisposableDemo\File.txt");
```
Comment (MithunSubramani) [inline src/assignment-8/IDisposableDemo/FileManager.cs:42]: use relative path.

File: src/assignment-8/IDisposableDemo/FileManager.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                string content = fileReader.ReadLine() !;
```
Comment (MithunSubramani) [inline src/assignment-8/IDisposableDemo/FileManager.cs:43]: Any reason to use null forgiving operator?

File: src/assignment-8/IDisposableDemo/FileManager.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                fileReader.Close();
+                this._fileWriter = new StreamWriter(@"D:\CSharp-Assignments\src\assignment-8\IDisposableDemo\File.txt");
```
Comment (MithunSubramani) [inline src/assignment-8/IDisposableDemo/FileManager.cs:45]: Why it the file is been closed here and file write instance is created?

**Review by MithunSubramani — CHANGES_REQUESTED** [review]

File: src/assignment-8/IDisposableDemo/FileManager.cs (hunk 1 above)
Comment (shafiq005-solin) [reply src/assignment-8/IDisposableDemo/FileManager.cs:57]: typos were fixed.

File: src/assignment-8/IDisposableDemo/FileManager.cs (hunk 3 above)
Comment (shafiq005-solin) [reply src/assignment-8/IDisposableDemo/FileManager.cs:43]: null forgiving operator removed.

File: src/assignment-8/IDisposableDemo/FileManager.cs (hunk 4 above)
Comment (shafiq005-solin) [reply src/assignment-8/IDisposableDemo/FileManager.cs:45]: If a Class is a child of I Disposable and objected were create in using block then, at the end of the using block it automatically call the Dispose method. To close the opened file automatically we have to keep the file open. That is the only reason to keep that open.

**Review by MithunSubramani — APPROVED** [review]

**Pull request merged**